### PR TITLE
Updated task service to use api v3

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes put_issue_url_here
### Changes

Updated task service and dependent controllers

---

UUID:

### CODE REVIEW

File: website/public/js/controllers/partyCtrl.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ habitrpg.controller("PartyCtrl", ['$rootScope','$scope','Groups','Chat','User','
         }
       }
 
-      Chat.markChatSeen($scope.group._id);
+      // Chat.markChatSeen($scope.group._id);
```
Comment (paglias) [inline website/public/js/controllers/partyCtrl.js]: ?

File: website/public/js/controllers/tasksCtrl.js [hunk 2]
```diff
@@ -24,7 +24,10 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
               if (direction === 'down') $rootScope.playSound('Minus_Habit');
               else if (direction === 'up') $rootScope.playSound('Plus_Habit');
       }
-      User.user.ops.score({params:{id: task.id, direction:direction}});
+      Tasks.scoreTask(task.id, direction)
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: These ops should use the shared code like User.user.ops.score so that we don't have to wait for the server response to update the UI

File: website/public/js/controllers/tasksCtrl.js [hunk 3]
```diff
@@ -33,11 +36,16 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
...
-        tags: _.transform(User.user.filters, function(m,v,k){
-          if (v) m[k]=v;
-        })
+        tags: _.transform(task.tags, function(m, v, k) {
+         if (v) m.push(v);
+        }),
       };
-      User.user.ops.addTask({body:newTask});
+
+      Tasks.createUserTasks(newTask)
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: same here (shared code)

File: website/public/js/controllers/tasksCtrl.js [hunk 4]
```diff
@@ -70,14 +78,18 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
...
+    };
 
     /**
      * Pushes task to top or bottom of list
      */
     $scope.pushTask = function(task, index, location) {
       var to = (location === 'bottom' || $scope.ctrlPressed) ? -1 : 0;
-      User.user.ops.sortTask({params:{id:task.id},query:{from:index, to:to}})
+      Tasks.moveTask(task.id, to);
+      User.user[task.type + 's'].splice(index, 1);
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: this is handled in shared code

File: website/public/js/controllers/tasksCtrl.js [hunk 5]
```diff
@@ -91,18 +103,27 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
       }
     };
 
-    $scope.removeTask = function(task) {
+    $scope.removeTask = function(task, index) {
       if (!confirm(window.env.t('sureDelete', {taskType: window.env.t(task.type), taskText: task.text}))) return;
-      User.user.ops.deleteTask({params:{id:task.id}})
+      User.user[task.type + 's'].splice(index, 1);
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: shared code

File: website/public/js/controllers/tasksCtrl.js [hunk 6]
```diff
@@ -157,50 +177,57 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
...
+    $scope.addChecklistItem = function(task, $event, $index) {
       if (!task.checklist[$index].text) {
         // Don't allow creation of an empty checklist item
         // TODO Provide UI feedback that this item is still blank
-      } else if ($index == task.checklist.length-1){
-        User.user.ops.updateTask({params:{id:task.id},body:task}); // don't preen the new empty item
-        task.checklist.push({completed:false,text:''});
-        focusChecklist(task,task.checklist.length-1);
+      } else if ($index == task.checklist.length - 1) {
+        Tasks.addChecklistItem(task.id, task.checklist[$index]);
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: I'll stop commenting on every function here but... shared code :)

File: website/public/js/controllers/tasksCtrl.js [hunk 7]
```diff
@@ -33,11 +36,16 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
       var newTask = {
         text: task,
         type: listDef.type,
-        tags: _.transform(User.user.filters, function(m,v,k){
-          if (v) m[k]=v;
-        })
+        tags: _.transform(task.tags, function(m, v, k) {
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: I this should rely on User.user.filters. it's used to add the selected tags to the newly created task

File: website/views/shared/tasks/edit/tags.jade [hunk 8]
```diff
@@ -1,5 +1,5 @@
 fieldset.option-group(ng-if='!$state.includes("options.social.challenges")')
   p.option-title.mega(ng-class='{active: task._tags}', ng-click='task._tags = !task._tags', tooltip=env.t('expandCollapse'))=env.t('tags')
   label.checkbox(ng-repeat='tag in user.tags', ng-if='task._tags')
-    input(type='checkbox', ng-model='task.tags[tag.id]')
+    input(type='checkbox', ng-model='task.tags[tag.id]', ng-checked="task.tags.indexOf(tag.id) !== -1")
```
Comment (paglias) [inline website/views/shared/tasks/edit/tags.jade:4]: why is this necessary?

Comment (TheHollidayInn) [inline website/views/shared/tasks/edit/tags.jade:4]: This is because tags are now an array and not an object

Comment (TheHollidayInn) [inline website/views/shared/tasks/edit/tags.jade:4]: I'll be able to remove `ng-model='task.tags[tag.id]` once I the userService is ported.

**Comment by TheHollidayInn** [comment]
Updated!

File: website/public/js/controllers/tasksCtrl.js [hunk 9]
```diff
@@ -91,18 +94,26 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
       }
     };
 
-    $scope.removeTask = function(task) {
+    $scope.removeTask = function(task, index) {
```
Comment (paglias) [inline website/public/js/controllers/tasksCtrl.js]: the index parameter is not used

**Comment by TheHollidayInn** [comment]
Updated!
